float64[40] x
float64[40] y
float64[40] z